docs/libcurl/opts/CURLOPT_READFUNCTION.3

```diff
@@ -64,6 +64,14 @@ server end and the transfer will end there.
 If you set this callback pointer to NULL, or don't set it at all, the default
 internal read function will be used. It is doing an fread() on the FILE *
 userdata set with \fICURLOPT_READDATA(3)\fP.
+
+When sending data to a remote server using a read callback, the application
+should tell libcurl the size of the upload data. This allows for better error
+checking, and in some cases it is vital to make the operation successful. If
+you do a HTTP POST, the size must be set with
+\fICURLOPT_POSTFIELDSIZE_LARGE(3)\fP or get the body sent using chunked
+encoding. If you set \fICURLOPT_UPLOAD(3)\fP, the size of the upload should
+get set with \fICURLOPT_INFILESIZE_LARGE(3)\fP.
```

Comment:
The size doesn't have to be set though. I would go with something shorter, like 
`You can set the total size of the data you are sending by using \fICURLOPT_INFILESIZE_LARGE(3)\fP or \fICURLOPT_POSTFIELDSIZE_LARGE(3)\fP, depending on the type of transfer. For some transfer types it may be required and also it allows for better error checking.`

Reply:
That's good, thanks!